samza-core/src/main/java/org/apache/samza/system/inmemory/InMemoryManager.java

```diff
@@ -71,7 +70,7 @@ void put(SystemStreamPartition ssp, Object key, Object message) {
     }
 
     IncomingMessageEnvelope messageEnvelope = new IncomingMessageEnvelope(ssp, offset, key, message,
-        0, 0L, Instant.now().toEpochMilli());
+        0, 0L, System.nanoTime());
```

Comment:
nanoTime != epoch millis in nanos.

Reply:
it doesn't matter since what we want to achieve at the end is to calculate the duration (latency) so we will compare 2 nanoTime points. The only case where this matter is when we read the eventTime from the kafka record timestamp, which is EpochMilli ... any suggestions how to handle this?

Reply:
this is resolved by reverting eventTime to its original semantics (i.e., epochMilli) and use nanoTime for arrivalTime (now renamed arrivalSystemNanoTime to be clear, and added javadoc)